src/Aspire.Dashboard/DashboardWebApplication.cs

```diff
@@ -437,18 +433,22 @@ await Microsoft.AspNetCore.Authentication.AuthenticationHttpContextExtensions.Si
         {
             _app.MapPost("/authentication/logout", () => TypedResults.SignOut(authenticationSchemes: [CookieAuthenticationDefaults.AuthenticationScheme, OpenIdConnectDefaults.AuthenticationScheme]));
         }
-    }
 
-    internal static string[] GetSupportedCultures()
-    {
-        var supportedCultures = CultureInfo.GetCultures(CultureTypes.AllCultures)
-            .Where(culture => LocalizedCultures.Contains(culture.TwoLetterISOLanguageName) || LocalizedCultures.Contains(culture.Name))
-            .Select(culture => culture.Name)
-            .ToList();
-
-        // Non-standard culture but it is the default in many Chinese browsers. Adding zh-CN allows OS culture customization to flow through the dashboard.
-        supportedCultures.Add("zh-CN");
-        return supportedCultures.ToArray();
+        _app.MapGet("/api/set-language", (string? language, string redirectUrl, HttpContext httpContext) =>
+        {
+            if (string.IsNullOrEmpty(language) || string.IsNullOrEmpty(redirectUrl))
+            {
+                return Task.FromResult(Results.BadRequest());
+            }
+
+            httpContext.Response.Cookies.Append(
+                CookieRequestCultureProvider.DefaultCookieName,
+                // keep current culture for formatting, etc., but change ui (display) culture
+                CookieRequestCultureProvider.MakeCookieValue(new RequestCulture(CultureInfo.CurrentCulture.Name, language)),
```

Comment:
Is it correct to keep the current culture for formatting? If you're viewing french language, shouldn't there also be french formatting of dates, numbers, etc?

Reply:
> shouldn't there also be french formatting of dates, numbers, etc

There's no "french" formatting, within each language it will vary by locale, ie fr-CH has some differences than fr-CA. We could add the ability to change the formatting culture as well, but since the current formatting behavior reflects what the user sees everywhere else, since it's their OS formatting culture, I think it's ok

Reply:
I know there are different French-like cultures. It was an example.

I think changing the language in the dropdown should be the same as changing the language in the browser. If fr-CH is selected from the `Accept-Language` value then both the language and number/date formatting are changed together.

Likewise, if a language is specified by `culture` query string value then it is set as both by default. They're only different if both `culture` and `ui-culture` query string values are specified.

Having two selections for language and formatting is overkill. I've never seen another website that does this. The question is, what is the behavior that most people expect and what, and I think it is changing them together is best.